components/ui/Button.tsx

```diff
@@ -70,22 +73,21 @@ export function CustomButton({
       style={[
         styles.buttonContainer,
         container,
-        disabled && styles.disabled,
+        loading ? styles.loading : disabled && styles.disabled,
         containerStyles,
       ]}
+      disabled={disabled || loading}
       {...rest}>
       <View style={children && title ? [styles.childrenContainer] : {}}>
         {children}
       </View>
-      {title && (
-        <CustomText
-          size={textSize}
-          fontWeight="semibold"
-          fontFamily="marfa"
-          style={[text, styles.buttonText, textStyles]}>
-          {title}
-        </CustomText>
-      )}
+      <CustomText
+        size={textSize}
+        fontWeight="semibold"
+        fontFamily="marfa"
+        style={[text, styles.buttonText, textStyles]}>
+        {loading ? "Saving..." : title}
```

Comment:
I meant to put the wording on the button itself because some button can have `saving...`, or `adding ...`.There is a possibility of loading text changing with the component where the button is needed